plugins/outputs/http/http.go

```diff
@@ -82,13 +82,27 @@ var sampleConfig = `
   ## Maximum amount of time before idle connection is closed.
   ## Zero means no limit.
   # idle_conn_timeout = 0
+
+  ## For status codes not in non_retryable_statuscodes list, number of times to retry
+  ## before metric buffer is discarded.
+  ## max_retries = 0, NEVER retry and discard the metric buffer for (<200 or >300); negates 
+  ## retryable_statuscodes list.
+  ## When not set or max_retries = -1, maintain default behaviour which is to continually retry.
+  # max_retries = 3
+
+  ## Optional list of status codes (<200 or >300) upon which requests should not be retried
+  # non_retryable_statuscodes = [400, 500]
+  
+  ## Specific set of status codes (<200 or >300) to retry.  All other status codes will cause
+  ## metric buffer to be discarded.
+  # retryable_statuscodes = [402, 503, 504]
 `
 
 const (
 	defaultContentType    = "text/plain; charset=utf-8"
 	defaultMethod         = http.MethodPost
 	defaultUseBatchFormat = true
-	defaultMaxRetries     = 0
+	defaultMaxRetries     = -1
```

Comment:
I assume that defaultMaxRetries is NEVER used?  Seems like we wouldn't ever want "retry forever" behavior based on feedback thus far.  Seems like defaultMaxRetries=0 is better to safe guard

Reply:
defaultMaxRetries = -1  is equivalent to not setting max_retries field in the telegraf config, which would match the original telegraf behaviour.  Recall, original telegraf did not have a max_retries config option.

Telegraf will continue to attempt to send the batch until it receives HTTP success.  If the buffer fills up because it keeps getting failed HTTP responses, it throws away the older metrics and adds the newer ones to the end of the buffer.  When that happens, the next "batch" telegraf attempts to send will have dropped the oldest metrics.